packages/editor-ui/src/stores/workflows.store.ts

```diff
@@ -273,6 +274,52 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, () => {
 
 	const getPastChatMessages = computed(() => Array.from(new Set(chatMessages.value)));
 
+	const dirtinessByName = computed(() => {
+		// Do not highlight dirtiness if new partial execution is not enabled
+		if (settingsStore.partialExecutionVersion === 1) {
+			return {};
+		}
+
+		const dirtiness: Record<string, CanvasNodeDirtiness | undefined> = {};
+		const visitedByName: Record<string, true | undefined> = {};
```

Comment:
Nit: Would using `Set` to track visited nodes here be clearer and type-safer?
```suggestion
		const visitedByName = new Set<string>();
```